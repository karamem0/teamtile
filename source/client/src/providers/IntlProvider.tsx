//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import {
  IntlProvider as Provider,
  createIntl,
  MessageFormatElement
} from 'react-intl';

import en from '../translations/compiled/en.json';
import ja from '../translations/compiled/ja.json';

const translations: { [key: string]: Record<string, MessageFormatElement[]> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language.substring(0, 2)]
});

function IntlProvider(props: React.PropsWithChildren<unknown>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
