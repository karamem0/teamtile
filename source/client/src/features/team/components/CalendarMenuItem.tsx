//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { app, appInstallDialog } from '@microsoft/teams-js';
import { useAsyncFn, useError } from 'react-use';
import Presenter from './CalendarMenuItem.presenter';
import { getTab } from '../managers/TeamManager';
import { isPC } from '../../../utils/Teams';
import messages from '../messages';
import { useIntl } from 'react-intl';
import { useSnackbar } from '../../../providers/SnackbarProvider';

interface CalendarMenuItemProps {
  id?: string,
  internalId?: string
}

function CalendarMenuItem(props: Readonly<CalendarMenuItemProps>) {

  const {
    id,
    internalId
  } = props;

  const { setSnackbar } = useSnackbar();
  const dispatchError = useError();
  const [ state, fetch ] = useAsyncFn((teamId: string, channelId: string, appId: string) => getTab(teamId, channelId, appId));
  const intl = useIntl();

  const handleClick = React.useCallback(async () => {
    if (id == null) {
      return;
    }
    if (internalId == null) {
      return;
    }
    if (isPC(await app.getContext())) {
      const appId = process.env.VITE_CALENDAR_APP_ID;
      const appTab = await fetch(id, internalId, appId);
      if (appTab?.webUrl) {
        await app.openLink(appTab.webUrl);
        return;
      }
      if (appInstallDialog.isSupported()) {
        await appInstallDialog.openAppInstallDialog({ appId });
        return;
      }
    }
    setSnackbar({
      intent: 'warning',
      text: intl.formatMessage(messages.OperationNotSupported)
    });
  }, [
    id,
    internalId,
    intl,
    fetch,
    setSnackbar
  ]);

  React.useEffect(() => {
    if (state.error == null) {
      return;
    }
    dispatchError(state.error);
  }, [
    state.error,
    dispatchError
  ]);

  return (
    <Presenter onClick={handleClick} />
  );

}

export default CalendarMenuItem;
