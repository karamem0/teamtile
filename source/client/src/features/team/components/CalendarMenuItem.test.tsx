//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render } from '@testing-library/react';

import IntlProvider from '../../../providers/IntlProvider';

import CalendarMenuItem from './CalendarMenuItem.presenter';

test('create shapshot of when loading is true', async () => {
  const params = {
    loading: true
  };
  const { asFragment } = render(
    <IntlProvider>
      <CalendarMenuItem {...params} />
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('create shapshot of when loading is false', async () => {
  const params = {
    loading: false
  };
  const { asFragment } = render(
    <IntlProvider>
      <CalendarMenuItem {...params} />
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
