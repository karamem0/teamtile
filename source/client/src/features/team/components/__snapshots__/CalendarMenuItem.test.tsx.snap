// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create shapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto auto;
  grid-gap: 0.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  cursor: pointer;
}

.emotion-1 {
  width: 1rem;
  height: 1rem;
}

<span
    aria-label="Open channel calendar"
    class="fui-Text emotion-0 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
    role="button"
  >
    <button
      class="fui-Button r1alrhcs ___cpe5h10_1yhfyj7 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f18ktai2 fwbmr0d f44c6la"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2"
      >
        <span
          aria-hidden="true"
          class="root_dd790ee3 emotion-1"
        >
          <svg
            class="svg_dd790ee3"
            focusable="false"
            viewBox="0 0 2048 2048"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M768 768h128v128H768V768zm384 768h128v128h-128v-128zm384-768h128v128h-128V768zm-384 0h128v128h-128V768zm-384 256h128v128H768v-128zm-384 0h128v128H384v-128zm1152 0h128v128h-128v-128zm-384 0h128v128h-128v-128zm-384 256h128v128H768v-128zm-384 0h128v128H384v-128zm1152 0h128v128h-128v-128zm-384 0h128v128h-128v-128zm-384 256h128v128H768v-128zm-384 0h128v128H384v-128zM2048 128v1792H0V128h384V0h128v128h1024V0h128v128h384zM128 256v256h1792V256h-256v128h-128V256H512v128H384V256H128zm1792 1536V640H128v1152h1792z"
            />
          </svg>
        </span>
      </span>
    </button>
  </span>
</DocumentFragment>
`;
